src/WebJobs.Extensions.DurableTask/DurableActivityContext.cs

```diff
@@ -102,6 +103,31 @@ internal object GetInput(Type destinationType)
             }
 
             string serializedValue = jToken.ToString(Formatting.None);
+
+            // Object inputs for out-of-proc activities are passed in their JSON-stringified form
+            // with a destination type of System.String. Unfortunately, deserializing a JSON string
+            // to a string causes MessagePayloadDataConverter to throw an exception. This is a
+            // workaround for that case.
+            if (destinationType.Equals(typeof(string)))
+            {
+                // Json.NET has no TryParse or IsValidJson method, so we'll optimistically parse
+                // and catch any exceptions (which shouldn't occur: all (in-proc; JSON and not JSON)
+                // or all other (out-of-proc: not-JSON) inputs with destination System.String should
+                // cast to valid JValues and be handled above.)
+                try

```

Comment:
I'm curious - is there a simple and fast test we can apply to the string, such as checking to see if the first character is a `{` or a `[`? I'm not sure if this would be useful or not since I don't yet fully understand how the current iteration works though...

Reply:
That might be better - or not necessary at all. My concern is how understandable our thought process would be, but I suppose that's what comments are for.

Currently, by my testing, here's how the possible cases behave:

In/Out of Proc | JSON?<sub>*</sub> | Destination Type 
--------------- | ----------------------- | -------------------
In-Proc | Yes | System.String 
In-Proc | No | System.String
Out-of-Proc | No | System.String

Reply:
I don't quite understand the meaning of the **JSON?** column. `serializedInput` is always JSON.  Do you actually mean **JObject** (as in a JSON object and not a JSON primitive like int or string)?

Following up on our in-person discussion, I'm starting to think that your proposal to do the following might just work:
```csharp
if (destinationType == typeof(string))
{
    return serializedValue;
}
```

The part I mentioned about string encoding actually doesn't apply because the actual JSON value we're trying to deserialize is not actually a string (otherwise we would have returned at ln 102 like you mentioned), but a JObject.

Reply:
JObject, yes.